src/solve/blackboxoptim.jl

```diff
@@ -1,16 +1,54 @@
+abstract type BBO end
+
+for j = string.(BlackBoxOptim.SingleObjectiveMethodNames)
```

Comment:
What's the motivation behind? I don't think it's a good idea.

Reply:
The motivation is that all optimisers are called via `algorithm()`. Thus, both `BlackBoxOptim` as well as `NLopt` have lines to achieve that for NLopt it is `(f::NLopt.Algorithm)=f` and for `BlackBoxOptim` its the code above.